src/Speckle.Sdk/Serialisation/Send/SendProcess.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Threading.Channels;
+using Open.ChannelExtensions;
+using Speckle.Sdk.Common;
+using Speckle.Sdk.Models;
+using Speckle.Sdk.Transports;
+using Speckle.Sdk.Transports.ServerUtils;
+
+namespace Speckle.Sdk.Serialisation.Send;
+
+public record SendProcessSettings(
+  int MaxSerializeThreads = 4,
+  int MaxObjectRequestSize = ServerApi.BATCH_SIZE_GET_OBJECTS,
+  int BatchWaitMilliseconds = 500
+);
+
+public sealed class SendProcess : IDisposable
+{
+  private readonly SendProcessSettings _settings = new();
+
+  public SendProcess(IModelTarget modelTarget)
+  {
+    SourceChannel = Channel.CreateUnbounded<Base>();
+    SendStage = new(modelTarget);
+    SerializeStage = new();
+  }
+
+  private long _requested;
+  private Serialized? _rootObjectSerialized;
+  private Base? _rootObject;
+  public Channel<Base> SourceChannel { get; }
+  public SerializeStage SerializeStage { get; set; }
+  public SendStage SendStage { get; set; }
+
+  public Action<ProgressArgs[]>? Progress { get; set; }
+
+  public void InvokeProgress() =>
+    Progress?.Invoke(
+      [
+        new ProgressArgs(ProgressEvent.UploadObject, _requested, null),
+        new ProgressArgs(ProgressEvent.SerializeObject, SerializeStage.Serialized, null),
+        new ProgressArgs(ProgressEvent.UploadBytes, SendStage.Sent, null)
+      ]
+    );
+
+  public async Task<(string rootObjId, IReadOnlyDictionary<string, ObjectReference> convertedReferences)> SaveObject(
```

Comment:
What's the Exception handling like for exceptions that occur inside of a Stage?

Do we need to review and itterate on this later inorder to have a well doucmented set of Exceptional cases that the Operations.Send2 and Operations.Receive2 functions will throw?